pkg/R/deserialize.R

```diff
@@ -1,9 +1,73 @@
-# Utility functions to serialize, deserialize etc.
+# Utility functions to deserialize objects from Java.
+
+# Type mapping from Java to R
+# 
+# void -> NULL
+# Int -> integer
+# String -> character
+# Boolean -> logical
+# Double -> numeric / double
+# Array[Byte] -> raw
+#
+# Array[T] -> list()
+# Object -> jobj
+
+readObject <- function(con) {
+  # Read type first
+  type <- readType(con)
+  readTypedObject(con, type)
+}
+
+readTypedObject <- function(con, type) {
+  switch (type,
+    "i" = readInt(con),
+    "c" = readString(con),
+    "b" = readBoolean(con),
+    "d" = readDouble(con),
+    "r" = readRaw(con),
+    "l" = readList(con),
+    "n" = NULL,
+    "j" = getJobj(readString(con)),
+    stop("Unsupported type for deserialization"))
+}
+
+readString <- function(con) {
+  stringLen <- readInt(con)
+  string <- readBin(con, raw(), stringLen, endian="big")
+  rawToChar(string)
+}
 
 readInt <- function(con) {
   readBin(con, integer(), n = 1, endian="big")
```

Comment:
So I thought about this in more detail and it turns out we don't really have a defined style we use consistently. I checked the Google R style guide and a bunch of R doc pages (like `?read.csv`, `?paste` etc.) and it looks the `arg = value` style is more commonly used. I switched all usages in SparkR to have the space between `arg` and `val`. Let me know if you have any thoughts on this